### DISCUSSION TITLE
Bevy Playground

### DISCUSSION THREAD

**Discussion by alice-i-cecile** [bevyengine/bevy#1128] [Ideas]
When writing examples and using Bevy as a teaching tool, it would be fantastic to have an interactive low-friction web-based terminal for Bevy. Unfortunately, [Rust playground](https://play.rust-lang.org/) doesn't work, because Bevy is not in the top 100 crates, and repl.it fails to build `ash`.

If we wanted this, we could:

1. Convince repl.it, rust playground or an alternative to support our use case.
2. Set up a custom Bevy playground on the website, and host our own version of Rust playground with the correct dependencies installed and a hello world project precompiled.

Graphics / audio support might be hard in this sort of environment; as tuboinatub mentioned `bevy_webgl2` might be a valid approach towards this.

**Comment by NiklasEi** [comment]
There now is https://learnbevy.com/playground [by `@Liam` on Discord](https://discord.com/channels/691052431525675048/692648638823923732/1191596430750666783)

**Comment by JimLynchCodes** [comment]
@NiklasEi Dang, that is a nice online editor and lots of examples! 

@alice-i-cecile I do still think trying to get it added to repl.it is a good idea. They have a cool live "mob programming" feature and a community of people coding in all different languages, but just having a bevy starter there could get more exposure to bevy and people experimenting with it.
